src/util/sss_ssh.c

```diff
@@ -240,8 +290,9 @@ sss_ssh_print_pubkey(struct sss_ssh_pubkey *pubkey, const char *keyhost)
         goto end;
     }
 
-    /* OpenSSH expects a linebreak after each key */
-    if (keyhost == NULL) {
+    /* Check if the host name part is included with the key.
+     * OpenSSH expects a linebreak after each key. */
+    if (keyhost == NULL || sss_ssh_key_has_host_name(repr, needlehost)) {
```

Comment:
(Not currently possible but) there is no guarantee that `needlehost != NULL` here.

Wouldn't it be possible to deduce `needlehost` from `keyhost` instead of new arg?

Reply:
Of course it can be done, but that means we need to call `parse_ssh_host()` again from this function, when it was already called from the caller function.

Reply:
Up to you